x-pack/plugins/security_solution/common/endpoint/types/trusted_apps.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { TypeOf } from '@kbn/config-schema';
+import { GetTrustedAppsRequestSchema } from '../schema/trusted_apps';
+import { ExceptionListItemSchema } from '../../../../lists/common';
+import { FoundExceptionListItemSchema } from '../../../../lists/common/schemas/response';
+
+/** API request params for retrieving a list of Trusted Apps */
+export type GetTrustedAppsListRequest = TypeOf<typeof GetTrustedAppsRequestSchema.query>;
+export type GetTrustedListAppsResponse = Pick<
+  FoundExceptionListItemSchema,
+  'per_page' | 'page' | 'total'
+> & {
+  data: TrustedApp[];
+};
+
+/** Type for a new Trusted App Entry */
+export type NewTrustedApp = Pick<
+  ExceptionListItemSchema,
+  'created_at' | 'created_by' | 'name' | 'description' | 'entries'
```

Comment:
re: casing - it comes down to what I have seen around in kibana: non code references are camelCased, everything else is snake_cased.

Re: 
> So sending entries means that we will have to have logic on the client that will have to translate the entries into a single field applicationPath.

Not sure I follow - why would you need logic on the client to translate?


Re: Type and pick from Exceptions
I think our data types should be defined to match the data storage as close as possible. In this case, because we (trusted apps and Exceptions) share the same storage schema, I feel its best to pick the properties that we will reuse from that schema, so that it it changes in the future to something that we don't support, we can get those (TS) errors surfaced at the earlier possible time.  This is also consistent with usage patterns I have seen across kibana.